test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -0,0 +1,133 @@
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
+	const (
+		instanceKey = "cloud.provider.com/instance"
+	)
+
+	var (
+		ns            *corev1.Namespace
+		defaultFlavor *kueue.ResourceFlavor
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		defaultFlavor = testing.MakeResourceFlavor("default").Label(instanceKey, "default").Obj()
+	})
+
+	ginkgo.When("Scheduling workloads on clusterQueues", func() {
+		var (
+			prodClusterQ *kueue.ClusterQueue
+			devClusterQ  *kueue.ClusterQueue
+			prodQueue    *kueue.LocalQueue
+			devQueue     *kueue.LocalQueue
+		)
+
+		ginkgo.BeforeEach(func() {
+			gomega.Expect(k8sClient.Create(ctx, defaultFlavor)).To(gomega.Succeed())
+
+			prodClusterQ = testing.MakeClusterQueue("prod-cq").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
+					Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodClusterQ)).Should(gomega.Succeed())
+
+			devClusterQ = testing.MakeClusterQueue("dev-cq").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
+					Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, devClusterQ)).Should(gomega.Succeed())
+
+			prodQueue = testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodQueue)).Should(gomega.Succeed())
+
+			devQueue = testing.MakeLocalQueue("dev-queue", ns.Name).ClusterQueue(devClusterQ.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, devQueue)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			framework.ExpectClusterQueueToBeDeleted(ctx, k8sClient, prodClusterQ, true)
+			framework.ExpectClusterQueueToBeDeleted(ctx, k8sClient, devClusterQ, true)
+			framework.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, defaultFlavor, true)
+		})
+
+		ginkgo.It("Should unblock admission of new workloads once the admitted workload is in PodsReady condition", func() {
+			ginkgo.By("checking the first prod workload gets admitted")
+			prodWl := testing.MakeWorkload("prod-wl", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
+
+			ginkgo.By("verifying that the second workload is waiting as the first one has PodsReady=False")
+			devWl := testing.MakeWorkload("dev-wl", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
```

Comment:
I think this test is too gentle:

```suggestion
			devWl := testing.MakeWorkload("dev-wl", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
			framework.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
```

Basically, create the workloads in rapid succession to increase the chances of race conditions.

In particular, this would ensure that `AssumeWorkload` also puts the workload in the list of unready workloads (which I think it does already).

Reply:
ack, also just added a new test which makes sure both workloads are considered in the same scheduling cycle

Reply:
done